### ISSUE TITLE
Clarify error message when an attribute can't be resolved

### ISSUE THREAD

**Issue by jskeet** [dotnet/roslyn#7360]
I've just been tearing my hair out fighting DNX etc - but a more helpful compiler error would have made everything clear.

Sample code:

```csharp
class Program
{
    [Foo]
    static void Main(string[] args) {}
}
```

Current error message:

> error CS0246: The type or namespace name 'Foo' could not be found (are you missing a using directive or an assembly reference?)

More useful error message:

> error CS0246: The type or namespace name 'Foo' or 'FooAttribute' could not be found (are you missing a using directive or an assembly reference?)

That may sound trivial, but now consider this example in `AssemblyInfo.cs`:

```csharp
[assembly: Guid(...)]
```

The error message saying that the type or namespace name `Guid` can't be found leads to trying to work out which dependency I would need to make `System.Guid` available - when *really* I need `System.Runtime.InteropServices.GuidAttribute`.

If I'd been paying closer attention to the location of the error rather than the error message I could have worked it out, certainly - but an error message which clarified that the compiler was really trying to find either of *two* type names would have got me there much, much faster.

**Comment by smudge202** [comment]
I've been caught out by this twice, but fortunately a Find all in Files found the culprit quickly for me (the only mention of `Guid` was the `AssemblyInfo` attribute).

**Comment by leppie** [comment]
@jskeet There you go :)

**Comment by VSadov** [comment]
Is there anything to be done for VB?

**Comment by leppie** [comment]
@VSadov Sorry I have absolutely no VB experience. Does VB also allow the shorthand attribute typenames? If so, the fix should be similar, if the legacy VB compiler was not strict.

**Comment by VSadov** [comment]
@leppie yes VB does that too. However it may also already implement this style of error reporting. At least we need to check the state of VB by verifying similar scenarios and if VB does not have same issues, then we can just close the bug.

**Comment by CyrusNajmabadi** [comment]
We now do report both:

```
The type or namespace name 'FooAttribute' could not be found (are you missing a using directive or an assembly reference?)
The type or namespace name 'Foo' could not be found (are you missing a using directive or an assembly reference?)
```
